src/box/lua/config/applier/credentials.lua

```diff
@@ -277,6 +277,101 @@ local function privileges_add_defaults(name, role_or_user, intermediate)
     return res
 end
 
+-- The table contains lists of all actions per object, that should be performed
+-- after object creation. Actions are listed in the following format:
+-- [obj_type][obj_name] = {{
+--     role_or_user,
+--     grant_or_revoke,
+--     user_or_role_name,
+--     privileges
+--   }, {
+--     ...
+--   },
+-- }
+local actions_on_creation = {
+    ['space'] = {},
+    ['function'] = {},
+    ['sequence'] = {},
+}
+
+-- Info about objects is stored inside system spaces
+-- box.space._space/_func/_sequence. Translate these names
+-- to obj_type values of space/function/sequence.
+local function translate_obj_type(obj_type)
+    if obj_type == '_space' then
+        return 'space'
+    elseif obj_type == '_func' then
+        return 'function'
+    elseif obj_type == '_sequence' then
+        return 'sequence'
+    end
+end
+
+-- Iterate through events and apply postponed action for newly created objects.
+local function on_commit_trigger(iterator)
+    log.debug('on_commit_trigger started')
+    for _, _, new_space, space_id in iterator() do
+        local obj_type = box.space[space_id].name
+        obj_type = translate_obj_type(obj_type)
+
+        -- Check if the event is for space/function/sequence.
+        if obj_type == nil then
+            goto skip
+        end
+
+        local obj_name = new_space.name
+        local obj_info = actions_on_creation[obj_type][obj_name] or {}
+
+        -- Apply all postponed actions for the object.
+        for _, info in ipairs(obj_info) do
+            local role_or_user, grant_or_revoke, name, privs = unpack(info)
+            -- There is a case when two objects have the same perms for the same
+            -- user and one is renamed to another, then the grant happens twice,
+            -- thus use if_not_exists = true.
+            box.schema[role_or_user][grant_or_revoke](name, privs, obj_type,
+                                            obj_name, {if_not_exists = true})
+
+            log.debug('credentials.apply: box.schema.%s.%s(%q, %q, %q, %q)',
+                      role_or_user, grant_or_revoke, name, privs,
+                      obj_type, obj_name)
+       end
+       actions_on_creation[obj_type][obj_name] = nil
+       ::skip::
+    end
+    log.debug('on_commit_trigger finished')
+end
+
+-- Check that event is space/function/sequence creation and set
+-- on_commit_trigger for the transaction.
+local function on_replace_trigger(_, new_space, obj_type, request_type)
```

Comment:
I guess that we don't want to attempt to perform DML in the read-only mode, so we should skip all the actions for 'applier' [session type](https://www.tarantool.io/en/doc/latest/reference/reference_lua/box_session/type/). It is a usual way to write on_replace/before_replace triggers.

Reply:
Done:
```diff
+      if box.session.type() == 'applier' then
+          -- The event is caused by replication, so the instance is in
+          -- RO mode. Thus, skip all applier actions.
+          return                                                                          
+      end